### ISSUE TITLE
Headers are being lowercased

### ISSUE THREAD

**Issue by bcooksey** [zapier/zapier-platform-core#13]
Setting a header in a `z.request` results in the header being lowercased when sent out on the wire.

Example code snippet to reproduce:

```javascript
  const promise = z.request(`https://api.lyft.com/oauth/token`, {
    method: 'POST',
    json: {
      code: bundle.inputData.code,
      grant_type: 'authorization_code'
    },
    headers: {
      'Content-Type': 'application/json',
      'Authorization': auth_header
    }
  });
```

Resulted in this request:

```
POST https://api.lyft.com/oauth/token
user-agent: Zapier
content-type: application/json; charset=utf-8
cache-control: no-cache
authorization: <auth_header>
```

**Comment by bryanhelmig** [comment]
This shouldn't matter - from https://www.w3.org/Protocols/rfc2616/rfc2616.html and newer revisions:

> Each header field consists of a name followed by a colon (":") and the field value. Field names are case-insensitive.

**Comment by BrunoBernardino** [comment]
It should not, but sadly it does for some APIs (not saying that's the case for Lyft). I believe we should preserve case.

**Comment by bcooksey** [comment]
As well intentioned as we were, #15 actually introduced a new bug.

Without all the headers being lowercased, the spots that try to add default headers (like [here](https://github.com/zapier/zapier-platform-core/blob/e201b775f953f1c7616859ce392864da8eb2263d/src/http-middlewares/before/prepare-request.js#L55)) now add duplicates like `{headers: {'Content-Type': '...', 'content-type': '...'}}`, which causes more issues than it solves.

Feel like we could:

1. Go back to lowercasing and say sorry to apps where it matters
2. Do the work to make our app casing agnostic, probably by using a helper anywhere we want to touch headers

**Comment by xavdid** [comment]
Hmm. On one hand, it's nice to conform to the standard (case doesn't matter) and have the onus be on the apis to also conform to that.

On the other hand, it would be doable to not wreck case. do a case-insensitive check to see if there's already a `content-type`, and if not add it (in whatever case). I like the idea of not changing user input out from under them, but i'm not sure if it's worth it.

**Comment by charettes** [comment]
FWIW Python's `requests` library work around this case issue by using a [case insensitive `dict` subclass](https://github.com/kennethreitz/requests/blob/b60c5a5ed46d75c56e36db5a58a98a50d880141d/requests/structures.py#L18-L89).

The idea is to use case normalized keys and store the case sensitive variant of the key along the value. I think that's how most the of CI stuff around do it (e.g. the `citext` extension of PostgreSQL).

**Comment by BrunoBernardino** [comment]
I'd vote on keeping the case people put, and checking if there's anything before really adding it to the headers (if there is an existing, case-insensitive match, remove it so the new key goes in instead).

**Comment by BrunoBernardino** [comment]
Will be fixed in 1.0.4
